addons/mail/models/mail_channel.py

```diff
@@ -88,21 +88,26 @@ def _generate_random_token(self):
         ('groups', 'Selected group of users')], string='Privacy',
         required=True, default='groups',
         help='This group is visible by non members. Invisible groups can add members through the invite button.')
-    group_public_id = fields.Many2one('res.groups', string='Authorized Group',
-                                      default=lambda self: self.env.ref('base.group_user'))
+    group_public_id = fields.Many2one('res.groups', string='Authorized Group', compute='_compute_group_public_id', readonly=False, store=True)
 
     _sql_constraints = [
         ('uuid_unique', 'UNIQUE(uuid)', 'The channel UUID must be unique'),
     ]
 
-    # CHAT CONSTRAINT
+    # CONSTRAINTS
 
     @api.constrains('channel_last_seen_partner_ids', 'channel_partner_ids')
     def _constraint_partners_chat(self):
         for ch in self.sudo().filtered(lambda ch: ch.channel_type == 'chat'):
             if len(ch.channel_last_seen_partner_ids) > 2 or len(ch.channel_partner_ids) > 2:
                 raise ValidationError(_("A channel of type 'chat' cannot have more than two users."))
 
+    @api.constrains('group_public_id', 'group_ids')
+    def _constraint_group_id_channel(self):
+        for channel in self.sudo().filtered(lambda channel: channel.channel_type != 'channel'):
+            if channel.group_public_id or channel.group_ids:
+                raise ValidationError(_('Group authorization and group auto-subscription are only supported on channels. Current channel type: %s', channel.channel_type))
```

Comment:
Your current error does not add valuable information as we don't know which channel is failing. Knowing the current channel type is not really interesting to fix the issue. I would say something like ``_("The following channels have group-based authorization but are not of right type 'channel': %(channel_names)s"), channel_names = ', '.join(failing_channel.mapped('name'))``